x-pack/test/observability_ai_assistant_functional/tests/contextual_insights/index.spec.ts

```diff
@@ -0,0 +1,163 @@
+      .service({ name: serviceName, environment: 'production', agentName: 'go' })
+      .instance('instance-a');
+
+    const interval = '1m';
+    const documents = [
+      timerange(start, end)
+        .interval(interval)
+        .rate(50)
+        .generator((timestamp) =>
+          serviceInstance
+            .transaction({ transactionName: 'GET /banana' })
+            .errors(
+              serviceInstance
+                .error({ message: 'Some exception', type: 'exception' })
+                .timestamp(timestamp)
+            )
+            .duration(10)
+            .timestamp(timestamp)
+            .failure()
+        ),
+    ];
+
+    await apmSynthtraceEsClient.index(documents);
+  }
+
+  async function createOpenAiConnector(proxy: LlmProxy) {
+    await supertest
+      .post('/api/actions/connector')
+      .set('kbn-xsrf', 'foo')
+      .send({
+        name: 'foo',
+        config: {
+          apiProvider: 'OpenAI',
+          apiUrl: `http://localhost:${proxy.getPort()}`,
+          defaultModel: 'gpt-4',
+        },
+        secrets: { apiKey: 'myApiKey' },
+        connector_type_id: '.gen-ai',
+      })
+      .expect(200);
+  }
+
+  async function deleteConnectors() {
+    const connectors = await supertest.get('/api/actions/connectors').expect(200);
+    const promises = connectors.body.map((connector: { id: string }) => {
+      return supertest
+        .delete(`/api/actions/connector/${connector.id}`)
+        .set('kbn-xsrf', 'foo')
+        .expect(204);
+    });
+
+    try {
+      await Promise.all(promises);
+    } catch (e) {
+      if (e?.response?.status === 404) {
+        return;
+      }
+
+      log.debug('Error deleting connector');
+      throw e;
+    }
+  }
+
+  async function navigateToError() {
+    await common.navigateToApp('apm', {
+      hash: `/services/opbeans-go/errors?rangeFrom=now-15m&rangeTo=now`,
+    });
+
+    await testSubjects.click('errorGroupId');
+  }
+
+  describe('Contextual insights for APM errors', () => {
+    let proxy: LlmProxy;
+
+    before(async () => {
+      await Promise.all([
+        deleteConnectors(), // cleanup previous connectors
+        apmSynthtraceEsClient.clean(), // cleanup previous synthtrace data
+        ui.auth.login(), // login
+      ]);
+
+      await generateErrorData();
+    });
+
+    describe('when there are no connectors', () => {
+      it('should not show the contextual insight component', async () => {
+        await navigateToError();
+        await retry.try(async () => {
+          await testSubjects.missingOrFail('obsAiAssistantInsightButton');
+        });
+      });
+    });
+
+    describe('when there are connectors', () => {
+      before(async () => {
+        proxy = await createLlmProxy(log);
+        await createOpenAiConnector(proxy);
+      });
+
+      it('should show the contextual insight component on the APM error details page', async () => {
+        await navigateToError();
+
+        proxy
```

Comment:
should this be awaited?

Reply:
No I don't think so. `await`'ing will block until the intercepter is called afair. I could do this instead:

```ts
       const waitForIntercept = proxy
          .intercept(
            'conversation',
            (body) => !isFunctionTitleRequest(body),
            'This error is nothing to worry about. Have a nice day!'
          )
          .complete();
```

and then instead of 

```ts
await proxy.waitForAllInterceptorsSettled();
```

I could do this:

```ts
await waitForIntercept;
```

Reply:
I feel that the ergonomics of `proxy.intercept` could be improved though. We already have a `waitForIntercept` method but only exposed if we don't call complete.

Reply:
Are you running into an error awaiting? I think you want the response to be completed before you run the assertion. `intercept(...).complete` is when you don't care about the intermediate state, which I think applies here.